src/nvim/mark.c

```diff
@@ -458,8 +458,33 @@ fmark_T *mark_get_local(buf_T *buf, win_T *win, int name)
     mark = mark_get_visual(buf, name);
     // previous context mark
   } else if (name == '\'' || name == '`') {
-    // TODO(muniter): w_pcmark should be stored as a mark, but causes a nasty bug.
-    mark = pos_to_mark(curbuf, NULL, win->w_pcmark);
```

Comment:
changing `curbuf` to `buf` doesn't work ?

Reply:
nope `'` mark is window-specific. Each window has its own w_pcmark, but nvim_buf_get_mark(buffer, "'") should return the mark from the window that displays the target buffer, not the current window.